spec/models/box_spec.rb

```diff
@@ -17,273 +11,96 @@
-      box.check_has_box_items # make sure there are items
-      # make sure at least one item needs research
-      create(:box_item, box: box, inventory_type: inventory_type_research_needed)
-      box.complete_design!
+      box = build(:box, :designed)
       box.researched_by_id = researcher.id;
-      box.save
       expect(box).to transition_from(:designed).to(:research_in_progress).on_event(:claim_research)
     end
 
     it "transitions from research_in_progress to researched" do
       time_now = Time.parse("Oct 11 2019")
       allow(Time).to receive(:now).and_return(time_now)
-
-      box_request_1.reviewed_by_id = reviewer.id;
-      box_request_1.save
-      box_request_1.claim_review!
-      box_request_1.complete_review!
-      box = box_request_1.box
-      box.designed_by_id = designer.id;
-      box.save
-      box.claim_design!
-      box.check_has_box_items # make sure there are items
-      expect(box).to receive(:send_research_solicitation_email!).and_return(true)
-
-      # make sure at least one item needs research
-      create(:box_item, box: box, inventory_type: inventory_type_research_needed)
-      box.complete_design!
-      box.researched_by_id = researcher.id;
-      box.save
-      box.claim_research!
+      box = build(:box, :research_in_progress)
       box.mark_box_items_as_researched!
       expect(box).to transition_from(:research_in_progress).to(:researched).on_event(:complete_research)
       expect(box.researched_at).to eq(time_now)
     end
 
     it "transitions from researched to assembly in progress" do
-      box_request_1.reviewed_by_id = reviewer.id;
-      box_request_1.save
-      box_request_1.claim_review!
-      box_request_1.complete_review!
-      box = box_request_1.box
-      box.designed_by_id = designer.id;
-      box.save
-      box.claim_design!
-      box.check_has_box_items # make sure there are items
-      # make sure at least one item needs research
-      create(:box_item, box: box, inventory_type: inventory_type_research_needed)
-      box.complete_design!
-      box.researched_by_id = researcher.id;
-      box.save
-      box.claim_research!
-      box.mark_box_items_as_researched!
-      box.complete_research!
+      box = build(:box, :researched)
       box.assembled_by_id = assembler.id;
-      box.save
       expect(box).to transition_from(:researched).to(:assembly_in_progress).on_event(:claim_assembly)
     end
 
-    it "transitons from assembly_in_progress to assembled" do
-      box_request_1.reviewed_by_id = reviewer.id;
-      box_request_1.save
-      box_request_1.claim_review!
-      box_request_1.complete_review!
-      box = box_request_1.box
+    it "transitions from assembly_in_progress to assembled" do
+      box = build(:box, :assembly_in_progress)
       allow(box).to receive(:send_shipping_solicitation_email!)
-      box.designed_by_id = designer.id;
-      box.save
-      box.claim_design!
-      box.check_has_box_items # make sure there are items
```

Comment:
Is this handled by any factories? Maybe the check_has_box_items should get added to the `designed` trait ?